test/CMakeLists.txt

```diff
@@ -1,36 +1,35 @@
-include_directories(${PROJECT_SOURCE_DIR}/src ${EXT_PROJECTS_DIR}/easyloggingpp.9.96.4 ${CATCH_INCLUDE_DIR})
-
-set(TEST_SRC_FILES
-    main.cpp
-    sample_catch_test.cpp
-    sample_yaml_cpp_test.cpp
-    #SimpleFakeItTest.cpp
-    Spatial/CoordinateTest.cpp
-    Spatial/LocationTest.cpp
-    Core/RandomTest.cpp
-    Core/TimeHelpersTest.cpp
-    Core/StringHelpersTest.cpp
-    Core/Config/ConfigTest.cpp
-    )
-
-add_executable(${PROJECT_TEST_NAME} ${TEST_SRC_FILES} )
-add_dependencies(${PROJECT_TEST_NAME} MaSimCore date args catch)
-
-target_link_libraries(${PROJECT_TEST_NAME}
-    MaSimCore
-    yaml-cpp
-    fmt::fmt fmt::fmt-header-only
-    GSL::gsl GSL::gslcblas
-    )
-
-target_compile_features(${PROJECT_TEST_NAME} PRIVATE cxx_range_for)
-
-
-add_custom_command(TARGET ${PROJECT_TEST_NAME} POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different
-    ${PROJECT_SOURCE_DIR}/misc/input.yml $<TARGET_FILE_DIR:${PROJECT_TEST_NAME}>)
-
-#
-#install(TARGETS ${PROJECT_TEST_NAME} DESTINATION ${PROJECT_SOURCE_DIR}/bin)
-#install(FILES ${PROJECT_SOURCE_DIR}/misc/input.yml DESTINATION ${PROJECT_SOURCE_DIR}/bin)
+# 'Google_test' is the subproject name
+find_package(fmt CONFIG REQUIRED)
+find_package(GTest CONFIG REQUIRED)
+find_package(GSL REQUIRED)
+#project(Google_tests)
+
+#set(SRC MDC/ModelDataCollector.cpp MDC/ModelDataCollector.h Spatial/Location.h
+#        Spatial/Location.cpp Spatial/Coordinate.h Spatial/Coordinate.cpp
+#        Helpers/TimeHelpers.h date/date.h Helpers/StringHelpers.h
+#         Helpers/NumberHelpers.h Core/Random.h Core/Random.cpp
+#        )
+
+set(SRC_TEST test1.cpp CoordinateTest.cpp LocationTest.cpp StringHelpersTest.cpp
+        TimeHelpersTest.cpp RandomTest.cpp ConfigTest.cpp Location.cpp Coordinate.cpp)
```

Comment:
Location.cpp, Coordinate.cpp should not be copied here, it should stay in 'src', you cannot modify /copy here

Reply:
If I am removing them then it gives error. If I put the path to these files as Spatial/Location.cpp and Spatial/Coordinate.cpp then also it gives error.

Reply:
In my old test set up, look for how I set up the dependency and link to MaSimCore

Reply:
I followed from your project and added MaSimCore in add_dependencies() and target_link_libraries() but it gives error

Reply:
Please more specifc on what error you got.

Reply:
@merlinvn Please have a look at this CMakeLists file for tests.

```
find_package(fmt CONFIG REQUIRED)
find_package(GTest CONFIG REQUIRED)
find_package(GSL REQUIRED)
#project(Google_tests)

include_directories(${PROJECT_SOURCE_DIR}/src ${EXT_PROJECTS_DIR}/easyloggingpp.9.96.4)


set(SRC_TEST test1.cpp CoordinateTest.cpp LocationTest.cpp core/StringHelpersTest.cpp
        core/TimeHelpersTest.cpp RandomTest.cpp ConfigTest.cpp)# Location.cpp Coordinate.cpp)

add_executable(main_test ${SRC_TEST})
add_dependencies(main_test date args MaSimCore)

include_directories (${POMS_SOURCE_DIR}/src)
include_directories(${EXT_PROJECTS_DIR}/easyloggingpp.9.96.4)


target_link_libraries(main_test PRIVATE MaSimCore yaml-cpp fmt::fmt-header-only
        GSL::gsl GSL::gslcblas GTest::gtest GTest::gtest_main)

target_compile_features(main_test PRIVATE cxx_range_for)


add_custom_command(TARGET main_test POST_BUILD
        COMMAND ${CMAKE_COMMAND} -E copy_if_different
        ${PROJECT_SOURCE_DIR}/misc/input.yml $<TARGET_FILE_DIR:main_test>)




```

Reply:
Please see this updated file:

```
# 'Google_test' is the subproject name
find_package(fmt CONFIG REQUIRED)
find_package(GTest CONFIG REQUIRED)
find_package(GSL REQUIRED)
#project(Google_tests)

include_directories(${POMS_SOURCE_DIR}/src ${EXT_PROJECTS_DIR}/easyloggingpp.9.96.4)


set(SRC_TEST test1.cpp CoordinateTest.cpp LocationTest.cpp core/StringHelpersTest.cpp
        core/TimeHelpersTest.cpp RandomTest.cpp ConfigTest.cpp)# Location.cpp Coordinate.cpp)

add_executable(main_test ${SRC_TEST})
add_dependencies(main_test date args MaSimCore)

#include_directories (${POMS_SOURCE_DIR}/src)
#include_directories(${EXT_PROJECTS_DIR}/easyloggingpp.9.96.4)

target_link_libraries(main_test PRIVATE MaSimCore yaml-cpp fmt::fmt-header-only
        GSL::gsl GSL::gslcblas GTest::gtest GTest::gtest_main)

target_compile_features(main_test PRIVATE cxx_range_for)


add_custom_command(TARGET main_test POST_BUILD
        COMMAND ${CMAKE_COMMAND} -E copy_if_different
        ${POMS_SOURCE_DIR}/misc/input.yml $<TARGET_FILE_DIR:main_test>)
```

Reply:
Look good. You should change the last line to match with the new "test_data/input.yml".
Currently your input.yaml is in the folder "test_data/misc", move it to "test_data" folder.

add_custom_command(TARGET main_test POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy_if_different ${POMS_SOURCE_DIR}/misc/input.yml $<TARGET_FILE_DIR:main_test>)

Reply:
I made changes and updated the CMakeLists file.  I deleted the build-debug and CMake created it. But I am getting some errors.
```
find_package(fmt CONFIG REQUIRED)
find_package(GTest CONFIG REQUIRED)
find_package(GSL REQUIRED)
#project(Google_tests)

include_directories(${POMS_SOURCE_DIR}/src ${EXT_PROJECTS_DIR}/easyloggingpp.9.96.4)


set(SRC_TEST test1.cpp CoordinateTest.cpp LocationTest.cpp core/StringHelpersTest.cpp
        core/TimeHelpersTest.cpp RandomTest.cpp ConfigTest.cpp)# Location.cpp Coordinate.cpp)

add_executable(main_test ${SRC_TEST})
add_dependencies(main_test date args MaSimCore)

#include_directories (${POMS_SOURCE_DIR}/src)
#include_directories(${EXT_PROJECTS_DIR}/easyloggingpp.9.96.4)

target_link_libraries(main_test PRIVATE MaSimCore yaml-cpp fmt::fmt-header-only
        GSL::gsl GSL::gslcblas GTest::gtest GTest::gtest_main)

target_compile_features(main_test PRIVATE cxx_range_for)


add_custom_command(TARGET main_test POST_BUILD
        COMMAND ${CMAKE_COMMAND} -E copy_if_different
        ${POMS_SOURCE_DIR}/test_data/input.yml $<TARGET_FILE_DIR:main_test>)
```

Reply:
![error_10](https://user-images.githubusercontent.com/20890066/110840938-5262f900-8273-11eb-8889-2757047ffb8c.JPG)

Reply:
When you delete this line "#set(CMAKE_CXX_FLAGS_DEBUG "-DNOMINMAX ${CMAKE_CXX_FLAGS_DEBUG} /MDd")", you didn't clean and build Cmakelist to see if it actually work or not i think....

Reply:
@merlinvn Thanks Nguyen. I have included this line and the error was gone. There is still one minor error. I have attached the image.

![bug](https://user-images.githubusercontent.com/20890066/110843868-ca7eee00-8276-11eb-8ea7-a8132983837d.JPG)

Reply:
I think you do not have the "input.yml" inside the "test_data" folder

Reply:
I did, I have added it to test data.

Reply:
This is the path to `input.yml`
`C:\Users\pua66\CLionProjects\PSU-CIDD-Malaria-Simulation\test\test_data\input.yml`

Reply:
Thanks. I got the error. It is working now.

Reply:
"${POMS_SOURCE_DIR}/test_data/input.yml" should be "${POMS_SOURCE_DIR}/test/test_data/input.yml"